tests/admin_utils/test_logentry.py

```diff
@@ -320,28 +408,29 @@ def test_action_flag_choices(self):
                 self.assertEqual(log.get_action_flag_display(), display_name)
 
     def test_hook_get_log_entries(self):
-        LogEntry.objects.log_action(
+        LogEntry.objects.log_actions(
             self.user.pk,
-            ContentType.objects.get_for_model(Article).pk,
-            self.a1.pk,
-            "Article changed",
+            [self.a1],
             CHANGE,
             change_message="Article changed message",
+            single_object=True,
         )
         c1 = Car.objects.create()
-        LogEntry.objects.log_action(
+        LogEntry.objects.log_actions(
             self.user.pk,
-            ContentType.objects.get_for_model(Car).pk,
-            c1.pk,
-            "Car created",
+            [c1],
             ADDITION,
             change_message="Car created message",
+            single_object=True,
         )
+        exp_str_article = escape(str(self.a1)[:200])
+        exp_str_car = escape(str(c1)[:200])
+
         response = self.client.get(reverse("admin:index"))
-        self.assertContains(response, "Article changed")
-        self.assertContains(response, "Car created")
+        self.assertContains(response, exp_str_article)
+        self.assertContains(response, exp_str_car)
 
         # site "custom_admin" only renders log entries of registered models
         response = self.client.get(reverse("custom_admin:index"))
-        self.assertContains(response, "Article changed")
-        self.assertNotContains(response, "Car created")
+        self.assertContains(response, exp_str_article)
+        self.assertNotContains(response, exp_str_car)
```

Comment:
There is no need to call `escape` and slice here, I would remove `exp_str_article`, `exp_str_car` 
```suggestion
        self.assertContains(response, str(self.a1))
        self.assertNotContains(response, str(c1))
```

Reply:
The test is failing if we don't add escape character

```
(endjango) C:\Users\akash\Desktop\Djangocon\django\tests>python runtests.py admin_utils.test_logentry.LogEntryTests.test_hook_get_log_entries
Testing against Django installed in 'C:\Users\akash\Desktop\Djangocon\django\django' with up to 8 processes
Found 1 test(s).
Creating test database for alias 'default'...
System check identified no issues (0 silenced).
F
======================================================================
FAIL: test_hook_get_log_entries (admin_utils.test_logentry.LogEntryTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\akash\Desktop\Djangocon\django\tests\admin_utils\test_logentry.py", line 436, in test_hook_get_log_entries
    self.assertContains(response, str(c1))
  File "C:\Users\akash\Desktop\Djangocon\django\django\test\testcases.py", line 539, in assertContains
    self.assertTrue(
AssertionError: False is not true : Couldn't find 'Car object (1)' in the following response
b'<!DOCTYPE html>\n\n<html lang="en" dir="ltr">\n<head>\n<title>Site administration | Django site admin</title>\n<link rel="stylesheet" href="/static/admin/css/base.css">\n\n  <link rel="stylesheet" href="/static/admin/css/dark_mode.css">\n  <script src="/static/admin/js/theme.js" defer></script>\n\n\n  <link rel="stylesheet" href="/static/admin/css/nav_sidebar.css">\n  <script src="/static/admin/js/nav_sidebar.js" defer></script>\n\n<link rel="stylesheet" href="/static/admin/css/dashboard.css">\n\n\n\n    <meta name="viewport" content="width=device-width, initial-scale=1.0">\n    <link rel="stylesheet" href="/static/admin/css/responsive.css">\n    \n\n<meta name="robots" content="NONE,NOARCHIVE">\n</head>\n\n<body class=" dashboard"\n  data-admin-utc-offset="-18000">\n<a href="#content-start" class="skip-to-content-link">Skip to main content</a>\n<!-- Container -->\n<div id="container">\n\n    \n    <!-- Header -->\n    \n      <header id="header">\n        <div id="branding">\n        \n<div id="site-name"><a href="/test_admin/custom_admin/">Django administration</a></div>\n\n\n        </div>\n        \n        \n        <div id="user-tools">\n            \n                Welcome,\n                <strong>super</strong>.\n            \n            \n                \n                    <a href="/">View site</a> /\n                \n                \n                    \n                    \n                \n                \n                <a href="/test_admin/custom_admin/password_change/">Change password</a> /\n                \n                <form id="logout-form" method="post" action="/test_admin/custom_admin/logout/">\n                    <input type="hidden" name="csrfmiddlewaretoken" value="n7oTvM4pCjEb0WwOX8ALOX9s5zA8AJxYusl9jYakzHt38OloSzYGu7H7GhqpUpIi">\n                    <button type="submit">Log out</button>\n                </form>\n                \n<button class="theme-toggle">\n  <div class="visually-hidden theme-label-when-auto">Toggle theme (current theme: auto)</div>\n  <div class="visually-hidden theme-label-when-light">Toggle theme (current theme: light)</div>\n  <div class="visually-hidden theme-label-when-dark">Toggle theme (current theme: dark)</div>\n  <svg aria-hidden="true" class="theme-icon-when-auto">\n    <use xlink:href="#icon-auto" />\n  </svg>\n  <svg aria-hidden="true" class="theme-icon-when-dark">\n    <use xlink:href="#icon-moon" />\n  </svg>\n  <svg aria-hidden="true" class="theme-icon-when-light">\n    <use xlink:href="#icon-sun" />\n  </svg>\n</button>\n\n            \n        </div>\n        \n        \n        \n      </header>\n    \n    <!-- END Header -->\n    \n    \n\n    <main class="main" id="main">\n      \n        \n      \n      <div id="content-start" class="content" tabindex="-1">\n        \n          \n        \n        <!-- Content -->\n        <div id="content" class="colMS">\n          \n          <h1>Site administration</h1>\n          \n          \n<div id="content-main">\n  \n\n\n  \n    <div class="app-admin_utils module">\n      <table>\n        <caption>\n          <a href="/test_admin/custom_admin/admin_utils/" class="section" title="Models in the Admin_Utils application">Admin_Utils</a>\n        </caption>\n        \n          <tr class="model-article">\n            \n              <th scope="row"><a href="/test_admin/custom_admin/admin_utils/article/">Articles</a></th>\n            \n\n            \n              <td><a href="/test_admin/custom_admin/admin_utils/article/add/" class="addlink">Add</a></td>\n            \n\n            \n              \n                <td><a href="/test_admin/custom_admin/admin_utils/article/" class="changelink">Change</a></td>\n              \n            \n          </tr>\n        \n      </table>\n    </div>\n  \n\n\n</div>\n\n          \n<div id="content-related">\n    <div class="module" id="recent-actions-module">\n        <h2>Recent actions</h2>\n        <h3>My actions</h3>\n            \n            \n            \n            <ul class="actionlist">\n            \n            <li class="changelink">\n                <span class="visually-hidden">Changed:</span>\n                \n                    <a href="/test_admin/admin/admin_utils/article/1/change/">Title</a>\n                \n                <br>\n                \n                    <span class="mini quiet">Article</span>\n                \n            </li>\n            \n            <li class="changelink">\n                <span class="visually-hidden">Changed:</span>\n                \n                    <a href="/test_admin/admin/admin_utils/article/1/change/">Title</a>\n                \n                <br>\n                \n                    <span class="mini quiet">Article</span>\n                \n            </li>\n            \n            </ul>\n            \n    </div>\n</div>\n\n          <br class="clear">\n        </div>\n        <!-- END Content -->\n        <div id="footer"></div>\n      </div>\n    </main>\n</div>\n<!-- END Container -->\n\n<!-- SVGs -->\n<svg xmlns="http://www.w3.org/2000/svg" class="base-svgs">\n  <symbol viewBox="0 0 24 24" width="1rem" height="1rem" id="icon-auto"><path d="M0 0h24v24H0z" fill="currentColor"/><path d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-2V4a8 8 0 1 0 0 16z"/></symbol>\n  <symbol viewBox="0 0 24 24" width="1rem" height="1rem" id="icon-moon"><path d="M0 0h24v24H0z" fill="currentColor"/><path d="M10 7a7 7 0 0 0 12 4.9v.1c0 5.523-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2h.1A6.979 6.979 0 0 0 10 7zm-6 5a8 8 0 0 0 15.062 3.762A9 9 0 0 1 8.238 4.938 7.999 7.999 0 0 0 4 12z"/></symbol>\n  <symbol viewBox="0 0 24 24" width="1rem" height="1rem" id="icon-sun"><path d="M0 0h24v24H0z" fill="currentColor"/><path d="M12 18a6 6 0 1 1 0-12 6 6 0 0 1 0 12zm0-2a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM11 1h2v3h-2V1zm0 19h2v3h-2v-3zM3.515 4.929l1.414-1.414L7.05 5.636 5.636 7.05 3.515 4.93zM16.95 18.364l1.414-1.414 2.121 2.121-1.414 1.414-2.121-2.121zm2.121-14.85l1.414 1.415-2.121 2.121-1.414-1.414 2.121-2.121zM5.636 16.95l1.414 1.414-2.121 2.121-1.414-1.414 2.121-2.121zM23 11v2h-3v-2h3zM4 11v2H1v-2h3z"/></symbol>\n</svg>\n<!-- END SVGs -->\n</body>\n</html>\n'

----------------------------------------------------------------------
Ran 1 test in 0.074s

FAILED (failures=1)
Destroying test database for alias 'default'...
```